### PR TITLE
Fix #4948: final implicit class can be overridden

### LINKED ISSUE #4948: final implicit class can be overridden
Overriding a final implicit class's equivalent definition shouldn't be possible.
```scala
  trait Foo
  trait Implicits {
    final implicit class FooExtender(foo : Foo) {
      def empty : Unit = {}
    }
  }

  object Test extends Implicits {
    override implicit def FooExtender(foo: Foo): Test.FooExtender = super.FooExtender(foo) //should not be allowed
  }
```
Same issue as in scalc https://github.com/scala/bug/issues/11078

### PR DESCRIPTION
**Pull request by Kobenko**
https://github.com/lampepfl/dotty/issues/4948

### CODE REVIEW

**Comment by Kobenko** [comment]
@allanrenucci yeah. sure.

**Comment by Kobenko** [comment]
@allanrenucci hi. can you check my test and tip me how correct to create CLA status? (it is strange, since I already signed it)

File: compiler/src/dotty/tools/dotc/ast/Desugar.scala [hunk 1]
```diff
@@ -293,7 +293,7 @@ object desugar {
     val impl @ Template(constr0, parents, self, _) = cdef.rhs
     val mods = cdef.mods
     val companionMods = mods
-        .withFlags((mods.flags & AccessFlags).toCommonFlags)
+        .withFlags((mods.flags & (AccessFlags | Flags.Final)).toCommonFlags)
```
Comment (allanrenucci) [inline compiler/src/dotty/tools/dotc/ast/Desugar.scala]:
```suggestion
        .withFlags((mods.flags & (AccessFlags | Final)).toCommonFlags)
```

File: tests/neg/overrideFinalImplicit.scala [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Implicits {
+  final implicit class FooExtender(foo: String)
+}
+
+class Test extends Implicits {
+  override implicit def FooExtender(foo: String) = super.FooExtender(foo) // error
+}
```
Comment (allanrenucci) [inline tests/neg/overrideFinalImplicit.scala]:
```suggestion
}

```

**Pull request merged**